packages/@lwc/synthetic-shadow/src/shared/event-target.ts

```diff
@@ -30,28 +51,18 @@ export function getEventListenerWrapper(fnOrObj: unknown) {
     let wrapperFn = EventListenerMap.get(fnOrObj);
     if (isUndefined(wrapperFn)) {
         wrapperFn = function (this: EventTarget, event: Event) {
+            const currentTarget = eventCurrentTargetGetter.call(event) as EventTarget;
+            const actualTarget = getActualTarget(event);
```

Comment:
@Sumitjg We need to investigate whether we're missing test coverage here because we weren't differentiating between hosts and roots.